samza-core/src/test/java/org/apache/samza/serializers/model/serializers/TestMetricsSnapshotSerdeV2.java

```diff
@@ -53,17 +91,44 @@ public void testSerde() {
     metricMessage.get(samzaContainerMetricsGroupName).put("exceptions", boundedList.getValues());
     metricMessage.get(samzaContainerMetricsGroupName).put("commit-calls", 0);
 
-    MetricsSnapshot metricsSnapshot = new MetricsSnapshot(metricsHeader, new Metrics(metricMessage));
+    String diagnosticsManagerGroupName = "org.apache.samza.diagnostics.DiagnosticsManager";
+    metricMessage.put(diagnosticsManagerGroupName, new HashMap<>());
+    metricMessage.get(diagnosticsManagerGroupName).put("containerCount", 1);
+    metricMessage.get(diagnosticsManagerGroupName).put("containerMemoryMb", 1024);
+    metricMessage.get(diagnosticsManagerGroupName).put("containerNumCores", 2);
+    metricMessage.get(diagnosticsManagerGroupName).put("numStores", 5);
 
+    Map<String, ContainerModel> containerModels = new HashMap<>();
+    Map<TaskName, TaskModel> tasks = new HashMap<>();
+
+    Set<SystemStreamPartition> sspsForTask1 = new HashSet<>();
+    sspsForTask1.add(new SystemStreamPartition("kafka", "test-stream", new Partition(0)));
+    tasks.put(new TaskName("Partition 0"), new TaskModel(new TaskName("Partition 0"), sspsForTask1, new Partition(0)));
+
+    Set<SystemStreamPartition> sspsForTask2 = new HashSet<>();
+    sspsForTask2.add(new SystemStreamPartition("kafka", "test-stream", new Partition(1)));
+    tasks.put(new TaskName("Partition 1"), new TaskModel(new TaskName("Partition 1"), sspsForTask2, new Partition(1)));
+
+    containerModels.put("0", new ContainerModel("0", tasks));
+
+    metricMessage.get(diagnosticsManagerGroupName).put("containerModels", containerModels);
+
+    MetricsSnapshot metricsSnapshot = new MetricsSnapshot(metricsHeader, new Metrics(metricMessage));
     MetricsSnapshotSerdeV2 metricsSnapshotSerde = new MetricsSnapshotSerdeV2();
     byte[] serializedBytes = metricsSnapshotSerde.toBytes(metricsSnapshot);
 
     MetricsSnapshot deserializedMetricsSnapshot = metricsSnapshotSerde.fromBytes(serializedBytes);
 
-    Assert.assertTrue("Headers map should be equal",
-        metricsSnapshot.getHeader().getAsMap().equals(deserializedMetricsSnapshot.getHeader().getAsMap()));
+    Assert.assertTrue(metricsSnapshot.getHeader().getAsMap().equals(deserializedMetricsSnapshot.getHeader().getAsMap()));
+  }
 
-    Assert.assertTrue("Metrics map should be equal",
-        metricsSnapshot.getMetrics().getAsMap().equals(deserializedMetricsSnapshot.getMetrics().getAsMap()));
+  HashMap convertMap(Map<String, Object> map) {
```

Comment:
1) Can make this a "private static" method.
2) What is this useful for? It looks like it is copying all maps (incl. nested) into `HashMap`s in order to generate the serialized JSON for the test. But in the prod scenario, this doesn't get used, so is the test adding something extra that shouldn't be there?